main/src/ca/uwaterloo/flix/language/phase/Eraser.scala

```diff
@@ -2,109 +2,309 @@ package ca.uwaterloo.flix.language.phase
 
 import ca.uwaterloo.flix.api.Flix
 import ca.uwaterloo.flix.language.CompilationError
-import ca.uwaterloo.flix.language.ast.ErasedAst.JType
 import ca.uwaterloo.flix.language.ast.ErasedAst.JType._
-import ca.uwaterloo.flix.language.ast.{ErasedAst, FinalAst}
-import ca.uwaterloo.flix.util.Validation
-import ErasedAst.{Expression => ErasedExp}
-import FinalAst.{Expression => FinalExp}
+import ca.uwaterloo.flix.language.ast.ErasedAst.{ErasedType, JType}
+import ca.uwaterloo.flix.language.ast.{ErasedAst, FinalAst, MonoType, Symbol}
+import ca.uwaterloo.flix.util.{InternalCompilerException, Validation}
+import ca.uwaterloo.flix.util.Validation._
 
-object Eraser {
+object Eraser extends Phase[FinalAst.Root, FinalAst.Root] {
 
-  def run(root: FinalAst.Root)(implicit flix: Flix): Validation[ErasedAst.Root, CompilationError] = flix.phase("Eraser") {
-    ???
-  }
+  def run(root: FinalAst.Root)(implicit flix: Flix): Validation[FinalAst.Root, CompilationError] = flix.phase("Eraser") {
```

Comment:
I would like to have all the helper functions move out and after the run function. These functions should be private.